src/main/java/duke/Parameter.java

```diff
@@ -0,0 +1,42 @@
+package duke;
+
+public class Parameter {
+    private final ParameterType type;
+    private final String value;
+
+    public Parameter(String input) throws IllegalParameterException {
+        String parameterType = input.split(" ", 2)[0];
+
+        switch (parameterType) {
+            case "by":
```

Comment:
There is indentation issue on this switch-case block too. Look at the comment above for more info.